packages/ensnode-utils/src/subname-helpers.ts

```diff
@@ -17,6 +21,65 @@ export const ROOT_NODE = namehash("");
 
 export const makeSubnodeNamehash = (node: Hex, label: Hex) => keccak256(concat([node, label]));
 
+// normalize address to match the reverse address format
+// as per https://docs.ens.domains/resolution/names#reverse-nodes
+const normalizedAddressDigits = (address: Hex): string => address.slice(2).toLowerCase();
+
+export interface LabelByReverseAddressArgs {
+  /** The reverse address */
+  reverseAddress: Hex;
+
+  /** The domain's labelhash */
+  labelhash: Labelhash;
+}
+
+/**
+ * Attempt to heal a labelhash to its original label using reverse address.
```

Comment:
```suggestion
 * Attempt to heal the labelhash of an addr.reverse subname using an address that might be related to the subname.
```

We don't perform this healing based on "reverse address". We do it based on "an address that might be related to the subname"